### PR TITLE
Add integration test for single-unit pod deletion

### LINKED ISSUE #96: Single unit deploy comes back in read-only mode after host reboot
### Bug Description

For a single unit deployment of the mysql-k8s operator, when the K8S its hosted on (in this case a single unit microk8s) is rebooted, the single mysql unit comes back up in read-only mode.

I dropped the debug log output as it appears that the charm thinks its operating in a clustered mode when its not.

### To Reproduce

juju deploy mysql-k8s 

<reboot microk8s>

mysql unit will come back up in read-only mode.

### Environment

microk8s - latest/stable
mysql-k8s - latest/edge

### Relevant log output

```shell
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-0/charm/lib/charms/mysql/v0/mysql.py", line 868, in get_cluster_primary_address
    output = self._run_mysqlsh_script("\n".join(get_cluster_primary_commands))
  File "/var/lib/juju/agents/unit-mysql-0/charm/src/mysqlsh_helpers.py", line 523, in _run_mysqlsh_script
    raise MySQLClientError(e.stderr)
charms.mysql.v0.mysql.MySQLClientError: Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory
verbose: 2022-10-06T09:37:11Z: Loading startup files...
verbose: 2022-10-06T09:37:11Z: Loading plugins...
verbose: 2022-10-06T09:37:11Z: About to connect to MySQL at: clusteradmin@mysql-0.mysql-endpoints.sunbeam.svc.cluster.local
verbose: 2022-10-06T09:37:11Z: About to connect to MySQL at: mysql://clusteradmin@mysql-0.mysql-endpoints.sunbeam.svc.cluster.local:3306?connect-timeout=5000
Traceback (most recent call last):
  File "<string>", line 2, in <module>
mysqlsh.Error: Shell Error (51314): Dba.get_cluster: This function is not available through a session to a standalone instance (metadata exists, instance belongs to that metadata, but GR is not active)


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./src/charm.py", line 478, in <module>
    main(MySQLOperatorCharm)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/main.py", line 431, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/main.py", line 142, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/framework.py", line 316, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/framework.py", line 784, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/framework.py", line 857, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-mysql-0/charm/lib/charms/data_platform_libs/v0/database_provides.py", line 187, in _on_relation_changed
    self.on.database_requested.emit(event.relation, app=event.app, unit=event.unit)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/framework.py", line 316, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/framework.py", line 784, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-mysql-0/charm/venv/ops/framework.py", line 857, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-mysql-0/charm/src/relations/database.py", line 86, in _on_database_requested
    primary_endpoint = self.charm._mysql.get_cluster_primary_address()
  File "/var/lib/juju/agents/unit-mysql-0/charm/lib/charms/mysql/v0/mysql.py", line 871, in get_cluster_primary_address
    raise MySQLGetClusterPrimaryAddressError(e.message)
charms.mysql.v0.mysql.MySQLGetClusterPrimaryAddressError: Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory
verbose: 2022-10-06T09:37:11Z: Loading startup files...
verbose: 2022-10-06T09:37:11Z: Loading plugins...
verbose: 2022-10-06T09:37:11Z: About to connect to MySQL at: clusteradmin@mysql-0.mysql-endpoints.sunbeam.svc.cluster.local
verbose: 2022-10-06T09:37:11Z: About to connect to MySQL at: mysql://clusteradmin@mysql-0.mysql-endpoints.sunbeam.svc.cluster.local:3306?connect-timeout=5000
Traceback (most recent call last):
  File "<string>", line 2, in <module>
mysqlsh.Error: Shell Error (51314): Dba.get_cluster: This function is not available through a session to a standalone instance (metadata exists, instance belongs to that metadata, but GR is not active)

unit-mysql-0: 10:37:11 ERROR juju.worker.uniter.operation hook "database-relation-changed" (via hook dispatching script: dispatch) failed: exit status 1
unit-mysql-0: 10:37:11 INFO juju.worker.uniter awaiting error resolution for "relation-changed" hook
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
Delete the pod in a single unit deployment and write data to new pod.

Closes #96

### CODE REVIEW

File: tests/integration/high_availability/test_self_healing.py [hunk 1]
```diff
@@ -407,3 +418,62 @@ async def test_graceful_full_cluster_crash_test(
...
+        )
+
+    # Write data
+    server_config_credentials = await get_server_config_credentials(unit)
+    primary_unit_address = await get_unit_address(ops_test, unit.name)
+    random_chars = generate_random_string(40)
+    create_records_sql = [
+        "CREATE DATABASE IF NOT EXISTS test",
+        "CREATE TABLE IF NOT EXISTS test.data_verification_table (id varchar(40), primary key(id))",
+        f"INSERT INTO test.data_verification_table VALUES ('{random_chars}')",
```
Comment (shayancanonical) [inline tests/integration/high_availability/test_self_healing.py:453]: there is a helper function `insert_data_into_mysql_and_validate_replication` and another one `clean_up_database_and_table` that could be used here

File: tests/integration/high_availability/test_self_healing.py [hunk 2]
```diff
@@ -407,3 +418,62 @@ async def test_graceful_full_cluster_crash_test(
 
     async with ops_test.fast_forward():
         await ensure_all_units_continuous_writes_incrementing(ops_test)
+
+
+@pytest.mark.order(7)
+@pytest.mark.abort_on_fail
+@pytest.mark.self_healing_tests
+async def test_single_unit_pod_delete(ops_test: OpsTest) -> None:
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/high_availability/test_self_healing.py:426]: i think so, it is testing the self-healing functionality after a single unit cluster's pod is deleted

File: tests/integration/high_availability/test_self_healing.py [hunk 3]
```diff
@@ -99,7 +110,7 @@ async def test_kill_db_process(ops_test: OpsTest, continuous_writes) -> None:
     await clean_up_database_and_table(ops_test, database_name, table_name)
 
 
-@pytest.mark.order(2)
+@pytest.mark.order(3)
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/high_availability/test_self_healing.py:113]: it is quite possible that all tests with the same `pytest.mark.order` are run alphabetically (im not sure). on second thought, i am ok with the change you've made if you'd like to stay with it

**Review by shayancanonical — APPROVED** [review]

File: tests/integration/high_availability/test_self_healing.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/high_availability/test_self_healing.py]: i see no reason then to use the `pytest-order` plugin (you're not missing something, I added the plugin not knowing that the tests were run in sequential order of definition)

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**